monitoring/local_monitoring/kafka_demo/kafka-demo.sh

```diff
@@ -0,0 +1,285 @@
+	# Modify kruize-clowdapp file to update the namespace and kafka bootstrap value
+	#######################################################################################
+
+	if [[ -f "$CLOWDAPP_FILE" ]]; then
+		echo -n "🔄 Modifying $CLOWDAPP_FILE..."
+
+		# Update the namespace value in the YAML file
+		sed -i "s/\(http:\/\/kruize-recommendations\.\)ephemeral-[a-z0-9]\+\(.*\)/\1${EPHEMERAL_NAMESPACE}\2/" "$CLOWDAPP_FILE"
+
+		# Update KAFKA_BOOTSTRAP_SERVERS value
+		sed -i "s/\(value: \"\)env-ephemeral-[a-z0-9]\+-[a-z0-9]\+-kafka-bootstrap\(\..*\)/\1${KAFKA_SVC_NAME}\2/" "$CLOWDAPP_FILE"
+		sed -i "s/\(value: \".*\)ephemeral-[a-z0-9]\+\(.*\)/\1${EPHEMERAL_NAMESPACE}\2/" "$CLOWDAPP_FILE"
+		echo "✅ Done!"
+	else
+		echo "Error: $CLOWDAPP_FILE not found. Skipping modification."
+	fi
+
+	echo -n "🔄 Deploying the application.Please wait..."
+	{
+		bonfire deploy ros-ocp-backend -C kruize-test
+	} >>"${LOG_FILE}" 2>&1
+	echo "✅ Installation complete!"
+
+	############################
+	# Expose Kruize svc
+	############################
+	echo "🔄 Exposing kruize-recommendations service..."
+	oc expose svc/kruize-recommendations || error_exit "Failed to expose kruize-recommendations service."
+	echo "✅ Done!"
+
+
+	########################
+	# Get the route
+	########################
+	KRUIZE_ROUTE=$(oc get route | grep kruize-recommendations-ephemeral | awk '{print $2}')
+	echo "KRUIZE_ROUTE = ${KRUIZE_ROUTE}"
+
+	################################
+	# Create Metric Profile
+	################################
+	echo -n "🔄 Creating Metric Profile..."
+	curl -X POST "http://${KRUIZE_ROUTE}/createMetricProfile" -d @resource_optimization_local_monitoring.json || error_exit "Failed to send createMetricProfile request."
+	echo "✅ Created Successfully!"
+
+	####################################################
+	# Invoke the Bulk Service and get the jobID
+	####################################################
+	echo "🔄 Invoking Bulk Service..."
+	echo "curl -s -X POST "${KRUIZE_ROUTE}/bulk" -H Content-Type: application/json -d '{\"datasource\":\"${DATASOURCE_NAME}\"}'"
+	curl -s -X POST "http://${KRUIZE_ROUTE}/bulk" -H "Content-Type: application/json" -d "{\"datasource\":\"${DATASOURCE_NAME}\"}" || error_exit "Failed to send bulk request."
+	echo "✅ Job_id generated!"
+
+	##########################################################################
+	# Start consuming the recommendations using recommendations-topic
+	##########################################################################
+	echo -n "🔄 Consuming recommendations from recommendations-topic..."
+	echo
+	KAFKA_POD_NAME=$(oc get pods | grep kafka | awk '{print $1}')
+	echo "KAFKA_POD_NAME = ${KAFKA_POD_NAME}"
+	echo "KAFKA_SVC_NAME = ${KAFKA_SVC_NAME}"
+	echo "EPHEMERAL_NAMESPACE = ${EPHEMERAL_NAMESPACE}"
+	echo "oc exec "$KAFKA_POD_NAME" -- bin/kafka-console-consumer.sh --topic recommendations-topic --bootstrap-server "${KAFKA_SVC_NAME}"."${EPHEMERAL_NAMESPACE}".svc.cluster.local:9092"
```

Comment:
@khansaad Please check if we can access the kafka bootstrap server from outside the kafka cluster on EE.

Reply:
Added the TLS certificate option to consume the messages remotely.
The Kafka server needs to have this listener added:
```
- name: route
     port: 9094
     tls: true
     type: route
```